frontend/src/components/review/CommentsPanel.tsx

```diff
@@ -34,17 +34,43 @@ const CommentsPanel = ({
 }: CommentPanelProps) => {
   const [comments, setComments] = useState<CommentResponse[]>([]);
   const [filterIndex, setFilterIndex] = useState(2);
+  const [threadHeadMap, setThreadHeadMap] = useState<boolean[]>([]);
 
   const filterOptions = ["All", "Resolved", "Unresolved"];
   const filterToResolved = [null, true, false];
   const commentsQuery = buildCommentsQuery(
     storyTranslationId,
     filterToResolved[filterIndex],
   );
+
+  const updateThreadHeadMap = (cmts: CommentResponse[]) => {
+    /* A comment is a thread head if
+     * * It has the smallest lineIndex of all present comments for a story line
+     * * It's thread predecessor is resolved
+     */
+    const newThreadHeadMap = cmts.map((c: CommentResponse, i: number) => {
+      if (
+        i ===
+        cmts.findIndex(
+          (_c) => _c.storyTranslationContentId === c.storyTranslationContentId,
+        )
+      )
+        return true;
+      const lastC = cmts[i - 1];
+      return (
+        !c?.resolved &&
+        lastC?.resolved &&
+        lastC?.storyTranslationContentId === c?.storyTranslationContentId
```

Comment:
just to confirm: this condition handles the case where the current comment thread corresponds to the same story line as the previous comment?

Reply:
yesssssssssssssss. if the previous comment is resolved but the current one is not, and they are both for the same content line, then the current one is the start of a new thread